build.gradle.kts

```diff
@@ -0,0 +1,92 @@
+plugins {
+    id("com.hivemq.extension")
+    id("com.github.hierynomus.license")
+    id("io.github.sgtsilvio.gradle.defaults")
+    id("org.asciidoctor.jvm.convert")
+}
+
+group = "com.hivemq.extensions"
+description = "HiveMQ Heartbeat Extension - Provides a readiness check via HTTP"
+
+hivemqExtension {
+    name.set("HiveMQ Heartbeat Extension")
+    author.set("HiveMQ")
+    priority.set(1000)
+    startPriority.set(1000)
+    sdkVersion.set("4.9.0")
+
+    resources {
+        from("LICENSE")
+    }
+}
+
+dependencies {
+    implementation("org.apache.commons:commons-lang3:${property("commons-lang.version")}")
+    implementation("org.apache.commons:commons-text:${property("commons-text.version")}")
+
+    implementation("jakarta.xml.bind:jakarta.xml.bind-api:${property("jakarta-xml-bind.version")}")
+    runtimeOnly("com.sun.xml.bind:jaxb-impl:${property("jaxb.version")}")
+
+    implementation("org.eclipse.jetty:jetty-server:${property("jetty.version")}")
+    implementation("org.eclipse.jetty:jetty-servlet:${property("jetty.version")}")
+}
+
+/* ******************** resources ******************** */
+
+val prepareAsciidoc by tasks.registering(Sync::class) {
+    from("README.adoc").into({ temporaryDir })
+}
+
+tasks.asciidoctor {
+    dependsOn(prepareAsciidoc)
+    sourceDir(prepareAsciidoc.map { it.destinationDir })
+}
+
+hivemqExtension.resources {
+    from("LICENSE")
+    from("README.adoc") { rename { "README.txt" } }
+    from("dns-discovery-diagram.png")
+    from(tasks.asciidoctor)
+}
+
+/* ******************** test ******************** */
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter-api:${property("junit-jupiter.version")}")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
```

Comment:
Can be replaced with testImplementation("org.junit.jupiter:junit-jupiter:${property("junit-jupiter.version")}")